work/v1/types.go

```diff
@@ -79,9 +79,55 @@ type ManifestConfigOption struct {
 	ResourceIdentifier ResourceIdentifier `json:"resourceIdentifier"`
 
 	// FeedbackRules defines what resource status field should be returned.
+	// +optional
+	FeedbackRules []FeedbackRule `json:"feedbackRules,omitempty"`
+
+	// UpdateStrategy defines the strategy to update this manifest
+	// optional
+	UpdateStrategy UpdateStrategy `json:"updateStrategy"`
+}
+
+// UpdateStrategy defines the strategy to update this manifest
+type UpdateStrategy struct {
+	// type defines the strategy to update this manifest, default value is Update.
+	// Update type means to update resource by an update call.
+	// CreateOnly type means do not update resource based on current manifest.
+	// ServerSideApply type means to update resource using server side apply with work-controller as the field manager.
+	// If there is conflict, the related Applied condition of manifest will be in the status of False with the
+	// reason of ApplyConflict.
+	// +kubebuilder:default=Update
+	// +kubebuilder:validation:Enum=Update;CreateOnly;ServerSideApply
 	// +kubebuilder:validation:Required
 	// +required
-	FeedbackRules []FeedbackRule `json:"feedbackRules"`
+	Type string `json:"type"`
+
+	// serverSideApply defines the configuration for server side apply. It is honored only when
+	// type of updateStrategy is ServerSideApply
+	// +optional
+	ServerSideApply *ServerSideApplyConfig `json:"serverSideApply,omitempty"`
+}
+
+type UpdateStrategyType string
+
+const (
+	// Update type means to update resource by an update call.
+	UpdateStrategyTypeUpdate UpdateStrategyType = "Update"
+
+	// CreateOnly type means do not update resource based on current manifest. This should be used only when
+	// ServerSideApply type is not support on the spoke, and the user on hub would like some other controller
+	// on the spoke to own the control of the resource.
+	UpdateStrategyTypeCreateOnly UpdateStrategyType = "CreateOnly"
+
+	// ServerSideApply type means to update resource using server side apply with work-controller as the field manager.
+	// If there is conflict, the related Applied condition of manifest will be in the status of False with the
+	// reason of ApplyConflict. This type allows another controller on the spoke to control certain field of the resource.
+	UpdateStrategyTypeServerSideApply UpdateStrategyType = "ServerSideApply"
+)
+
+type ServerSideApplyConfig struct {
```

Comment:
Do you want to provide a way to control the agent string as well?  Some kind of suffix perhaps rather than allowing complete control.

Once this capability exists, it becomes possible to have two manifestworks touching different parts of the same resource.

Reply:
yes, add an fieldManager field with prefix validation